AsyncDisplayKit/Private/ASDisplayNode+UIViewBridge.mm

```diff
@@ -37,29 +40,46 @@
 #define __loaded (_layer != nil)
 
 #if DISPLAYNODE_USE_LOCKS
-#define _bridge_prologue ASDisplayNodeAssertThreadAffinity(self); ASDN::MutexLocker l(_propertyLock)
+#define _bridge_prologue_read ASDN::MutexLocker l(_propertyLock); ASDisplayNodeAssertThreadAffinity(self)
+#define _bridge_prologue_write ASDN::MutexLocker l(_propertyLock)
 #else
-#define _bridge_prologue ASDisplayNodeAssertThreadAffinity(self)
+#define _bridge_prologue_read ASDisplayNodeAssertThreadAffinity(self)
+#define _bridge_prologue_write
 #endif
 
+/// Returns YES if the property set should be applied to view/layer immediately.
+/// Side Effect: Registers the node with the shared ASPendingStateController if
+/// the property cannot be immediately applied and the node does not already have pending changes.
+/// This function must be called with the node's lock already held (after _bridge_prologue_write).
+ASDISPLAYNODE_INLINE BOOL ASDisplayNodeShouldApplyBridgedWriteToView(ASDisplayNode *node) {
+  if (ASDisplayNodeThreadIsMain()) {
```

Comment:
It might be worth adding an assertion here that checks that the lock is held. It is definitely a good idea to avoid creating another Mutex locker here, even though they are recursive, because even with the lock held this stack object would be unfortunate overhead